src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -81,15 +100,24 @@ object Scalajsld {
         .text("Execute the specified main(Array[String]) method on startup")
       opt[File]('o', "output")
         .valueName("<file>")
-        .required()
-        .action { (x, c) => c.copy(output = x) }
-        .text("Output file of linker (required)")
+        .action { (x, c) => c.copy(output = Some(x)) }
+        .text("Output file of linker (deprecated)")
+      opt[File]("outputDir")
+        .valueName("<file>")
+        .action { (x, c) => c.copy(outputDir = Some(x)) }
+        .text("Output directory of linker (required)")
       opt[Unit]('f', "fastOpt")
         .action { (_, c) => c.copy(noOpt = false, fullOpt = false) }
         .text("Optimize code (this is the default)")
       opt[Unit]('n', "noOpt")
         .action { (_, c) => c.copy(noOpt = true, fullOpt = false) }
         .text("Don't optimize code")
+      opt[ModuleSplitStyle]("moduleSplitStyle")
+        .action { (x, c) => c.copy(moduleSplitStyle = x) }
+        .text("Module splitting strategy " + ModuleSplitStyleRead.All.mkString("(", ", ", ")"))
+      opt[String]("outputPatterns")
+        .action { (x, c) => c.copy(outputPatterns = OutputPatterns.fromJSFile(x)) }
+        .text("Creates output patterns from a JS file pattern.")
```

Comment:
I think this help text can be improved. Currently, it describes what the flag does from the perspective of the CLI implementation. It should describe what the flag does from the perspective of the CLI user (i.e. change the names of the files that are written).

Further, it should point to documentation explaining what a valid pattern / value is (other texts might also fall short of this, depending on how much scopt auto-generates; feel free to improve that in a separate commit / PR).

Reply:
Honestly, I'm not familiar with ScalaJS, specially it's internals. I understand `outputPatterns` form CLI perspective, as target filenames template.

Maybe """File pattern used to create output files, and relative URI pattern, By default "%s.js""""? I feel like I'm not having right context. @gzm0

Reply:
How about: "Pattern for JS file names (default: `%s.js`). Expects a printf-style pattern with a single placeholder for the module ID. A typical use case is changing the file extension, e.g. `%.mjs` for Node.js modules."?